File: app/src/main/java/edu/aku/hassannaqvi/tpvics_hh/contracts/VersionAppContract.java
package edu.aku.hassannaqvi.tpvics_hh.contracts;

import android.database.Cursor;
import android.provider.BaseColumns;

import org.json.JSONException;
import org.json.JSONObject;

public class VersionAppContract {

    private static final String TAG = VersionAppContract.class.getName();
    String versioncode;
    String versionname;
    String pathname;

    public VersionAppContract() {
        // Default Constructor
    }

    public VersionAppContract Sync(JSONObject jsonObject) throws JSONException {
        this.versioncode = jsonObject.getString(VersionAppTable.COLUMN_VERSION_CODE);
        this.pathname = jsonObject.getString(VersionAppTable.COLUMN_PATH_DUPLICATE);
        this.versionname = jsonObject.getString(VersionAppTable.COLUMN_VERSION_NAME);
        return this;
    }

    public void hydrate(Cursor cursor) {
        this.versioncode = cursor.getString(cursor.getColumnIndex(VersionAppTable.COLUMN_VERSION_CODE));
        this.pathname = cursor.getString(cursor.getColumnIndex(VersionAppTable.COLUMN_PATH_NAME));
        this.versionname = cursor.getString(cursor.getColumnIndex(VersionAppTable.COLUMN_VERSION_NAME));
    }

    public String getVersioncode() {
        return versioncode;
    }

    public void setVersioncode(String versioncode) {
        this.versioncode = versioncode;
    }

    public String getPathname() {
        return pathname;
    }

    public void setPathname(String pathname) {
        this.pathname = pathname;
    }

    public String getVersionname() {
        return versionname;
    }

    public void setVersionname(String versionname) {
        this.versionname = versionname;
    }

    public static abstract class VersionAppTable implements BaseColumns {

        public static final String TABLE_NAME = "versionApp";
        public static final String COLUMN_VERSION_PATH = "elements";
        public static final String COLUMN_VERSION_CODE = "versionCode";
        public static final String COLUMN_VERSION_NAME = "versionName";
        public static final String COLUMN_PATH_NAME = "path";
        public static final String COLUMN_PATH_DUPLICATE = "outputFile";

        public static final String _URI = "output.json";
    }
}

File: app/src/main/java/edu/aku/hassannaqvi/tpvics_hh/contracts/VersionAppContract.kt
package edu.aku.hassannaqvi.tpvics_hh.contracts

import android.database.Cursor
import android.provider.BaseColumns
import org.apache.commons.lang3.StringUtils
import org.json.JSONException
import org.json.JSONObject

class VersionAppContract {
    var versioncode = StringUtils.EMPTY
    var versionname = StringUtils.EMPTY
    var pathname = StringUtils.EMPTY

    @Throws(JSONException::class)
    fun sync(jsonObject: JSONObject): VersionAppContract {
        versioncode = jsonObject.getString(VersionAppTable.COLUMN_VERSION_CODE)
        pathname = jsonObject.getString(VersionAppTable.COLUMN_PATH_NAME)
        versionname = jsonObject.getString(VersionAppTable.COLUMN_VERSION_NAME)
        return this
    }

    fun hydrate(cursor: Cursor): VersionAppContract {
        versioncode = cursor.getString(cursor.getColumnIndex(VersionAppTable.COLUMN_VERSION_CODE))
        pathname = cursor.getString(cursor.getColumnIndex(VersionAppTable.COLUMN_PATH_NAME))
        versionname = cursor.getString(cursor.getColumnIndex(VersionAppTable.COLUMN_VERSION_NAME))
        return this
    }

    object VersionAppTable : BaseColumns {
        const val TABLE_NAME = "versionApp"
        const val COLUMN_ID = "_id"
        const val COLUMN_VERSION_PATH = "elements"
        const val COLUMN_VERSION_CODE = "versionCode"
        const val COLUMN_VERSION_NAME = "versionName"
        const val COLUMN_PATH_NAME = "outputFile" //previous = path
        const val _URI = "output.json"
    }
}